examples/xoxo/xoxo.py

```diff
@@ -0,0 +1,183 @@
+from pysmt.shortcuts import FreshSymbol, Symbol, String, LE, GE, Int, Not, Or, And, Equals, Plus, Solver, StrContains
+from pysmt.typing import INT, STRING
+from enum import Enum
+import logging
+logger = logging.getLogger('xoxo')
+logger.setLevel(logging.DEBUG)
+ch = logging.StreamHandler()
+ch.setLevel(logging.INFO) # change to DEBUG for info on assertions
+logger.addHandler(ch)
+
+class Cell(Enum):
+    x = 1
+    o = 10
+    s = 0
+
+DEBUG = False
+turns = 0
+sq_size = 3
+test = 'tests/test1.txt'
+test = 'tests/blank.txt'
+
+# create symbols for the board
+board = [[FreshSymbol(INT) for _ in xrange(sq_size)]
+    for _ in xrange(sq_size)]
+
+solver = Solver(name="z3")
+
+# setup assertions for board
+for row in board:
+    for cell in row:
+        solver.add_assertion(Or([Equals(cell, Int(i.value))
+             for i in Cell]))
+
+display_board = [[Cell.s for _ in xrange(sq_size)] for _ in xrange(sq_size)]
+
+# load board
+with open(test) as fh:
+    for row, line in enumerate(fh.readlines()):
+        for col, cell in enumerate(line.strip().split(' ')):
+            if cell == Cell.x.name:
+                solver.add_assertion(Equals(board[row][col], Int(Cell.x.value)))
+                display_board[row][col] = Cell.x
```

Comment:
I would separate the two concerns. First, load the board and obtain an up-to-date version of the board. Then you can extract relevant constraints for the solver.